### PR TITLE
[WIP] feat(gms): Adds custom ownership types

### PR DESCRIPTION
**Pull request by pedro93**
# Summary
Adds a new entity to DataHub's metadata model graph: `OwnershipType` which will allow DataHub Admins to specify their own ownership types beyond and possibly in place of the existing ones: `Technical Owner`, `Business Owner`, `Data Steward`, `None`

## Details:
- Adds a new bootstrap step which will create the existing ownership types: `Technical Owner`, `Business Owner`, `Data Steward`, `None` as new entity instances of the new OwnershipType entity.
- Adds a new flag `INGEST_DEFAULT_OWNERSHIP_TYPES`, set to `true` by default which will control if the default ownership types should be ingested. If set to true, this ingestion process will only happen once.

## Checklist

- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

**Review by mmmeeedddsss — COMMENTED** [review]
(I'm not from the core datahub team, the pr is shared in https://datahubspace.slack.com/archives/C02QMLWJG12/p1679611195773309)

I assume you are already planning but this change will require some additional tweaks at least on the frontend. Hope these will not be that much.

Great PR with a huge value, I'm very excited to see it on action !

File: metadata-models/src/main/pegasus/com/linkedin/common/Owner.pdl [hunk 1]
```diff
@@ -26,8 +26,19 @@ record Owner {
...
 
+  /**
+   * The type of the ownership
+   * Urn of type O
+   */
+  @Relationship = {
+    "name": "ownershipType",
+    "entityTypes": [ "ownershipType" ]
+  }
+  customType: optional Urn
```
[the remark this answers is not in the document]
Comment (pedro93) [reply metadata-models/src/main/pegasus/com/linkedin/common/Owner.pdl:40]: Hello @mmmeeedddsss this PR includes a DataHub Bootstrap step that will populate the database with new ownership type entity instances based on the existing non-deprecated ownership types.

[the remark this answers is not in the document]
Comment (mmmeeedddsss) [reply metadata-models/src/main/pegasus/com/linkedin/common/Owner.pdl:40]: @pedro93 As far as I can see, these custom values are not used in anywhere. Do you plan to open a separate pr for the usage of these? (Such as graphql changes, frontend changes etc)

File: metadata-models/src/main/pegasus/com/linkedin/ownership/OwnershipTypeInfo.pdl [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+import com.linkedin.common.AuditStamp
+
+/**
+ * Information about an ownership type
+ */
+@Aspect = {
+  "name": "ownershipTypeInfo"
+}
+record OwnershipTypeInfo {
```
[the remark this answers is not in the document]
Comment (pedro93) [reply metadata-models/src/main/pegasus/com/linkedin/ownership/OwnershipTypeInfo.pdl:11]: This aspect is merely an information view of the new entity that this PR introduces. The urn is in the key aspect.

**Comment by pedro93** [comment]
Closing PR as it is incomplete, feature development will continue in a fork and will be made available once finished.

**Comment by ragerin** [comment]
@pedro93 where can we follow the fork for this feature?

**Pull request closed without merging**